packages/axios-extension/src/factory.ts

```diff
@@ -80,23 +86,53 @@ export function createForAbap(config: AxiosRequestConfig & Partial<ProviderConfi
     return createInstance(AbapServiceProvider, config);
 }
 
+export enum Authentication {
+    OAuth = 'OAuth',
+    ReentranceTicket = 'ReentranceTicket'
+}
+
+export interface AbapEnvironmentBtp {
+    authentication: Authentication.OAuth;
+    service: ServiceInfo;
+    refreshToken?: string;
+    refreshTokenChangedCb?: RefreshTokenChanged;
+}
+
+export interface AbapEnvironmentCloud {
+    authentication: Authentication.ReentranceTicket;
+    url: string;
+}
+
+export type AbapEnvironment = AbapEnvironmentBtp | AbapEnvironmentCloud;
+
 /**
- * Create an instance of an ABAP service provider for an ABAP environment on SAP BTP.
+ * Create an instance of an ABAP service provider for a Cloud ABAP system.
  *
- * @param service ABAP environment service
- * @param refreshToken optional refresh token
- * @param refreshTokenChangedCb option callback for refresh token updates
+ * @param options
  * @returns instance of an ABAP service provider
  */
-export function createForAbapOnBtp(
-    service: ServiceInfo,
-    refreshToken?: string,
-    refreshTokenChangedCb?: RefreshTokenChanged
-): AbapServiceProvider {
-    const provider = createInstance<AbapServiceProvider>(AbapServiceProvider, {
-        baseURL: service.url
-    });
-    attachUaaAuthInterceptor(provider, service, refreshToken, refreshTokenChangedCb);
+export function createForAbapOnCloud(options: AbapEnvironment & Partial<ProviderConfiguration>): AbapServiceProvider {
+    let provider: AbapServiceProvider;
+    switch (options.authentication) {
+        case Authentication.OAuth:
+            const { service, refreshToken, refreshTokenChangedCb } = options;
+            provider = createInstance<AbapServiceProvider>(AbapServiceProvider, {
+                baseURL: service.url
+            });
+            attachUaaAuthInterceptor(provider, service, refreshToken, refreshTokenChangedCb);
+            break;
+        case Authentication.ReentranceTicket:
+            const { url, ...config } = options;
```

Comment:
Where is `config ` coming from? I am lost here.

Reply:
Is this a generic "give me all properties" and then in line 128 you say "spread all properties"? If yes, why wouldn't it be the same for both cases?

Reply:
`config` is the type `ProviderConfiguration - (AbapEnvironmentCloud | AbapEnvironmentBtp)`.
I didn't use it in both cases as I moved the BTP code here - didn't want to break something that was already existing :)